backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointEventTypeRepository.java

```diff
@@ -107,6 +107,57 @@ public void updateEventTypesLinkedToEndpoint(UUID endpointId, Set<UUID> eventTyp
         entityManager.merge(endpoint);
     }
 
+    @Transactional
+    public void refreshEndpointLinksToEventType(String orgId, List<UUID> endpointsList) {
+        if (endpointsList == null || endpointsList.isEmpty()) {
+            return;
+        }
+
+        String deleteQueryStr = "DELETE FROM EndpointEventType eet WHERE eet in " +
+            "(from EndpointEventType where " +
+                "endpoint.orgId " + (orgId == null ? "is null " : "=: orgId ") +
+                "and id.endpointId in (:endpointList))";
```

Comment:
```suggestion
        String deleteQueryStr = "DELETE FROM EndpointEventType eet WHERE " +
                "endpoint.orgId " + (orgId == null ? "is null " : "=: orgId ") +
                "and id.endpointId in (:endpointList)";
```

Reply:
I failed when I tried something like this, but probably because I was attempting to fetch `endpoint` myself.
Your solution works, so Thank you.